File: android/feature/shop/src/main/java/com/enterprise/feature/shop/domain/entities/ShopState.kt
package com.enterprise.feature.shop.domain.entities

import com.enterprise.core.domain.entities.CartItem
import com.enterprise.core.domain.entities.DomainError
import java.math.BigDecimal

/** Shopping cart state for MVI pattern. */
data class ShopState(
    val cartItems: List<CartItem> = emptyList(),
    val cartItemCount: Int = 0,
    val isProcessing: Boolean = false,
    val error: DomainError? = null
) {
    val totalPrice: BigDecimal
        get() = cartItems.fold(BigDecimal.ZERO) { acc, item -> acc + item.totalPrice }
}

File: ios/EnterpriseMobileApp/Sources/Feature/Shop/Domain/Entities/ShopState.swift
import Foundation
import CoreDomain

/// Shopping feature state.
public struct ShopState: Equatable, Sendable {
    public var cartItems: [CartItem]
    public var cartItemCount: Int
    public var isProcessing: Bool
    public var error: DomainError?

    public init(
        cartItems: [CartItem] = [],
        cartItemCount: Int = 0,
        isProcessing: Bool = false,
        error: DomainError? = nil
    ) {
        self.cartItems = cartItems
        self.cartItemCount = cartItemCount
        self.isProcessing = isProcessing
        self.error = error
    }

    public var totalPrice: Decimal {
        cartItems.reduce(Decimal.zero) { $0 + $1.totalPrice }
    }
}
